LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/data/MyLocationManager.kt

```diff
@@ -0,0 +1,109 @@
+import com.google.android.gms.location.sample.locationupdatesbackgroundkotlin.hasPermission
+import java.util.concurrent.TimeUnit
+
+private const val TAG = "MyLocationManager"
+
+/**
+ * Manages all location related tasks for the app.
+ */
+class MyLocationManager private constructor(private val context: Context) {
+
+    val trackingLocation: MutableLiveData<Boolean> by lazy {
+        MutableLiveData<Boolean>(false)
+    }
+
+    // The Fused Location Provider provides access to location tracking APIs.
+    private val fusedLocationClient: FusedLocationProviderClient by lazy {
+        LocationServices.getFusedLocationProviderClient(context)
+    }
+
+    // Stores parameters for requests to the FusedLocationProviderApi.
+    private val locationRequest: LocationRequest by lazy {
+        LocationRequest()
+            // Sets the desired interval for active location updates. This interval is inexact. You
+            // may not receive updates at all if no location sources are available, or you may
+            // receive them slower than requested. You may also receive updates faster than
+            // requested if other applications are requesting location at a faster interval.
+            //
+            // IMPORTANT NOTE: Apps running on "O" devices (regardless of targetSdkVersion) may
+            // receive updates less frequently than this interval when the app is no longer in the
+            // foreground.
+            .setInterval(TimeUnit.SECONDS.toMillis(60))
+
+            // Sets the fastest rate for active location updates. This interval is exact, and your
+            // application will never receive updates faster than this value.
+            .setFastestInterval(TimeUnit.SECONDS.toMillis(30))
+
+            // Sets the maximum time when batched location updates are delivered. Updates may be
+            // delivered sooner than this interval.
+            .setMaxWaitTime(TimeUnit.MINUTES.toMillis(2))
+
+            .setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY)
+    }
+
+    private val locationUpdatePendingIntent: PendingIntent by lazy {
+        // API level 26 and above (Oreo+) place limits on Services, so we use a BroadcastReceiver.
+        val intent = Intent (context, LocationUpdatesBroadcastReceiver::class.java)
+        intent.action = LocationUpdatesBroadcastReceiver.ACTION_PROCESS_UPDATES
+        PendingIntent.getBroadcast(context, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT)
+    }
+
+    fun startLocationUpdates() {
+        Log.d(TAG, "startLocationUpdates()")
+
+        if (!context.hasPermission(Manifest.permission.ACCESS_FINE_LOCATION)) return
+
+        try {
+            trackingLocation.value = true
+            fusedLocationClient.requestLocationUpdates(locationRequest, locationUpdatePendingIntent)
+
+        } catch (e: SecurityException) {
+            trackingLocation.value = false
+            e.printStackTrace()
```

Comment:
nit: Consider a better name(?) for 'e' and maybe writing the error + stack trace to logcat with a more meaningful tag?

Reply:
+1 to Nicole. 
Also, since this is a sample for location, would it be worth saying something about when this might throw?

Reply:
Done... let me know if that comment looks ok.